src/components/ComponentDef.js

```diff
@@ -37,13 +37,7 @@ ComponentDef.prototype = {
      * so that we can instantiate nested components before their parents.
      */
     ___addChild: function (componentDef) {
-        var children = this.___children;
-
-        if (children) {
-            children.push(componentDef);
-        } else {
-            this.___children = [componentDef];
-        }
+        (this.___children || (this.___children = [])).push(componentDef);
```

Comment:
This change might have an impact on performance since it is hot code. My concern is that creating an array in one operation and then pushing to it in another operation might be slower. It is worth running some benchmarks to compare performance.